test/monitoring/control-plane_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build packet
+// +build prometheus
+
+package monitoring
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+	"time"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/portforward"
+	"k8s.io/client-go/transport/spdy"
+)
+
+func portForward(t *testing.T, readyChan, stopChan chan struct{}, closeChan func()) {
+	defer closeChan()
+
+	kubeconfig := testutil.KubeconfigPath(t)
```

Comment:
I think this function should be refactored. Building config is not part of the port forwarding. Also service and ports should be taken as an arguments, so any kind of service and ports can be forwarded. That might be useful other functionality.

This will also make easier testing this function.